### DISCUSSION TITLE
Using MkDocs from Github Actions with Material - Plugins not enabled?

### DISCUSSION THREAD

**Discussion by dcaddick** [mkdocs/mkdocs#3103] [Q&A]
Hi, first time posting here in Github Discussions.

I have managed to use MkDocs & Material via Github Actions to publish a website to Github pages.
And I will freely admit that I'm very dangerous and chiefly code with a chainsaw and sledge hammer...

What I'm finding difficult to understand is how the Plugins work?
In my Github workflow I have understood enough to install plugins via pip here:
[MakeDocsAction.yaml](https://github.com/dcaddick/gsd_public/blob/main/.github/workflows/MakeDocsAction.yaml)
But also curious as to whether or not I should be just able to "add" these to the list via the requirements.txt file?
[requirements.txt](https://github.com/dcaddick/gsd_public/blob/main/requirements.txt)

Anyway - getting them installed is one thing, easy enough, the hard part that I'm struggling with is getting them enabled?
See here for the mkdocs.yml in root:
[mydocs.yml](https://github.com/dcaddick/gsd_public/blob/main/mkdocs.yml)

The only ones I can get running are Search, Simple & Macros - am I doing something really dumb? 
Snippet:
```
# Plugins
plugins:
  - search
  - simple
  #- techdocs-core
  #- glightbox
  #- social
  #- awesome-list
  #- table-reader
  #- img2fig
  #- mermaid2
  - macros
  # - minify
```

**Comment by oprypin** [comment]
The problem with your example of GitHub actions is that it `uses: athackst/mkdocs-simple-plugin@v2.1.3` - most likely that isolates the environment from whatever you installed in the prior step. Instead of that whole section, just `run: mkdocs build` etc.

Where did you find the example that suggests doing this?

**Reply by dcaddick** [reply]
This was simply one of the "GitHub Actions" that was in the marketplace that I stumbled on and to date it works for me.
I'm guessing there are quite a number of different ways of achieving this?

**Comment by Andre601** [comment]
From what I can see are you not even installing the material theme?

It's not a theme bundled with MkDocs itself (Rather the oposite: It bundles MkDocs with it alongside other dependencies), so you have to `pip install` it yourself.

Also, unless you have parts of your docs inside actual code (therefore no external/separate markdown pages), using the simple plugin is kind of useless here and won't really help improve things.
If you only have separate doc pages, use the normal `mkdocs build` and an action like [`peaceiris/actions-gh-pages`](https://github.com/peaceiris/actions-gh-pages) to push your changes to your gh pages branch and/or folder.

Here is an example I use for my stuff (The final step can be simplified to a simple `run: mkdocs gh-deploy` if you use a `gh-pages` branch):
https://github.com/Andre601/blog/blob/master/.github/workflows/deploy.yml
